yarn-project/telemetry-client/src/otel.ts

```diff
@@ -135,27 +140,39 @@ export class OpenTelemetryClient implements TelemetryClient {
   public static async createAndStart(config: TelemetryClientConfig, log: Logger): Promise<OpenTelemetryClient> {
     const resource = await getOtelResource();
 
+    const tracer = config.useGcloudObservability
```

Comment:
I'm wondering if it would be better to pass in a factory, rather than parsing this config here? Then we parse the config at the top level and create the required factory.